codec-http2/src/test/java/io/netty/handler/codec/http2/Http2FrameCodecTest.java

```diff
@@ -209,6 +209,19 @@ public void headerRequestHeaderResponse() throws Exception {
         assertTrue(channel.isActive());
     }
 
+     @Test
+     public void flowControlShouldBeResilientToMissingStreams() throws Http2Exception {
+         Http2Connection conn = new DefaultHttp2Connection(true);
+         Http2ConnectionEncoder enc = new DefaultHttp2ConnectionEncoder(conn, new DefaultHttp2FrameWriter());
+         Http2ConnectionDecoder dec = new DefaultHttp2ConnectionDecoder(conn, enc, new DefaultHttp2FrameReader());
+         Http2FrameCodec codec = new Http2FrameCodec(enc, dec, new Http2Settings());
+         EmbeddedChannel em = new EmbeddedChannel(codec);
+
+         // We call #consumeBytes on a stream id which has not been seen yet to emulate the case
+         // where a stream is deregistered which in reality can happen in response to a RST.
+         assertFalse(codec.consumeBytes(1, 1));
```

Comment:
You will also need to call em.finishAndReleaseAll() (and assert the return value of it). Otherwise the test leaks buffers